### PR TITLE
Thunks: Adds functional thunk testing to CI

### PR DESCRIPTION
**Pull request by Sonicadvance1**
This is the bare minimum, it only tests glxinfo and vulkaninfo with and
without thunks. Nothing more special than that. Already found the .1 bug
with libvulkan host library loading.

### CODE REVIEW

**Comment by Sonicadvance1** [comment]
Interesting failure in CI here. I've seen this before but I thought it went away. Will need to look at why vulkaninfo crashes after doing stuff.

**Comment by Sonicadvance1** [comment]
The broadcom vulkan driver is breaking if it is installed on non-broadcom hardware. trying to find a way to cleanly resolve. This is where this problem comes from.

File: unittests/ThunkFunctionalTests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+function(AddThunksTest Bin ThunksFile)
+  string(LENGTH "${ThunksFile}" ThunkFileLength)
+
+  set (ARGS
+    "-t" "${CMAKE_INSTALL_PREFIX}/lib/fex-emu/HostThunks"
+    "-j" "${CMAKE_INSTALL_PREFIX}/share/fex-emu/GuestThunks"
+    "-o" "stderr" "--no-silent" "-c" "irjit" "-n" "500"
+    )
+  if (ThunkFileLength EQUAL 0)
```
Comment (neobrain) [inline unittests/ThunkFunctionalTests/CMakeLists.txt:11]: CMake is a really good scripting language (/s), so this works:
```suggestion
  if (NOT ThunkFile)
```

File: unittests/CMakeLists.txt [hunk 2]
```diff
@@ -6,10 +6,15 @@ add_subdirectory(POSIX/)
...
 if (BUILD_THUNKS)
   add_subdirectory(ThunkLibs)
 endif()
 
 if (BUILD_FEX_LINUX_TESTS)
   add_subdirectory(FEXLinuxTests/)
 endif()
+
+if (BUILD_THUNKS)
+  add_subdirectory(ThunkFunctionalTests)
```
Comment (neobrain) [inline unittests/CMakeLists.txt:19]: Why not move this into the same scope as ThunkLibs for grouping?

File: unittests/ThunkFunctionalTests/CMakeLists.txt (hunk 1 above)
Comment (Sonicadvance1) [reply unittests/ThunkFunctionalTests/CMakeLists.txt]: Oh right. Resolved.

File: unittests/CMakeLists.txt (hunk 2 above)
Comment (Sonicadvance1) [reply unittests/CMakeLists.txt]: Resolved.

**Comment by Sonicadvance1** [comment]
Had to update the CI rootfs to get the thunk tests running. Now shipping a rootfs to the CI machines that includes mesa from https://gitlab.freedesktop.org/sonicadvance1/mesa/-/commits/fex_ci

This also unveiled some weaknesses in the script that CI runs for updating the rootfs, so that needs to live here as well. It now deletes the squashfs image on error to be more robust to problems, and also deletes the extracted folder to ensure permissions errors don't occur on update.

**Review by neobrain — APPROVED** [review]
Worth noting `vulkaninfo` fails to exit successfully (both with and without thunks) in Parallels VMs on M1, but we can merge this since the CI machines are fine.

**Pull request merged**